core/app/c/[communitySlug]/stages/[stageId]/page.tsx

```diff
@@ -0,0 +1,80 @@
+import { Suspense } from "react";
+import Link from "next/link";
+import { notFound } from "next/navigation";
+
+import type { StagesId } from "db/public";
+import { Capabilities } from "db/src/public/Capabilities";
+import { MembershipType } from "db/src/public/MembershipType";
+import { Button } from "ui/button";
+
+import { CreatePubButton } from "~/app/components/pubs/CreatePubButton";
+import { getPageLoginData } from "~/lib/authentication/loginData";
+import { userCan } from "~/lib/authorization/capabilities";
+import { findCommunityBySlug } from "~/lib/server/community";
+import { getCommunityStages } from "~/lib/server/stages";
+import { PubListSkeleton } from "../../pubs/PubList";
+import { StagePubs } from "../components/StageList";
+
+export default async function Page({
+	params,
+	searchParams,
+}: {
+	searchParams: Record<string, string> & { page?: string };
+
+	params: { communitySlug: string; stageId: StagesId };
+}) {
+	const { communitySlug, stageId } = params;
+	const [{ user }, community] = await Promise.all([
+		getPageLoginData(),
+		findCommunityBySlug(communitySlug),
+	]);
+
+	if (!community) {
+		notFound();
+	}
+
+	const stage = await getCommunityStages({ stageId }).executeTakeFirstOrThrow();
+	const page = searchParams.page ? parseInt(searchParams.page) : 1;
+
+	const showEditButton = await userCan(
+		Capabilities.editCommunity,
+		{ type: MembershipType.community, communityId: community.id },
+		user.id
+	);
```

Comment:
since these queries don't depend on each other, could we wrap them in a `Promise.all`?